pkg/applicationserver/io/mqtt/format_json.go

```diff
@@ -16,7 +16,9 @@ package mqtt
 
 import (
 	"go.thethings.network/lorawan-stack/pkg/applicationserver/io/formatters"
-	"go.thethings.network/lorawan-stack/pkg/applicationserver/io/mqtt/topics"
+	mqtt "go.thethings.network/lorawan-stack/pkg/applicationserver/io/mqtt/topics"
+	"go.thethings.network/lorawan-stack/pkg/applicationserver/io/pubsub"
+	"go.thethings.network/lorawan-stack/pkg/applicationserver/io/pubsub/topics"
```

Comment:
What is this and why do we use it here?

Reply:
I've decoupled the already existing MQTT topic structure and format (both the `Layout` and `Format` interfaces) such that we can keep the same part structure (i.e. NATS has `v3.app1.devices.dev1.down.push` and MQTT has `v3/app1/devices/dev1/down/push`) and formats.

The `pubsub/topics` package contains the `Layout` interface that we had for MQTT before, and provides an `V3` topic layout with an no-op `AcceptedTopic`. The `mqtt` topics / format are just built on top of this interface and extend the generic layout.

I did this in order to maintain the topic structure across different PubSub implementations, without just duplicating the code (since it the `v3` topic layout would have been 99% the same, with `AcceptedTopic` being the only difference.